<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>bitFlagRendererDockWidget</class>
 <widget class="QgsDockWidget" name="bitFlagRendererDockWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>368</width>
    <height>364</height>
   </rect>
  </property>
  <property name="windowIcon">
   <iconset resource="../resources/bitflagrenderer.qrc">
    <normaloff>:/bitflagrenderer/icons/bitflagimage.png</normaloff>:/bitflagrenderer/icons/bitflagimage.png</iconset>
  </property>
  <property name="windowTitle">
   <string>Bit Flag Renderer</string>
  </property>
  <widget class="QWidget" name="dockWidgetContents">
   <layout class="QFormLayout" name="formLayout">
    <property name="horizontalSpacing">
     <number>2</number>
    </property>
    <property name="verticalSpacing">
     <number>2</number>
    </property>
    <property name="leftMargin">
     <number>2</number>
    </property>
    <property name="topMargin">
     <number>2</number>
    </property>
    <property name="rightMargin">
     <number>2</number>
    </property>
    <property name="bottomMargin">
     <number>2</number>
    </property>
    <item row="0" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>Layer</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QgsMapLayerComboBox" name="cbLayer">
      <property name="allowEmptyLayer">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>Band</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QgsRasterBandComboBox" name="cbBand"/>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>Cursor</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QLineEdit" name="lineEdit">
      <property name="readOnly">
       <bool>false</bool>
      </property>
      <property name="placeholderText">
       <string>Cursor location bits</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0" colspan="2">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>2</number>
      </property>
      <item>
       <widget class="QToolButton" name="btnRemoveParameters">
        <property name="text">
         <string>-</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../QGIS/images/images.qrc">
          <normaloff>:/images/themes/default/mActionRemove.svg</normaloff>:/images/themes/default/mActionRemove.svg</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnAddParameter">
        <property name="text">
         <string>+</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../QGIS/images/images.qrc">
          <normaloff>:/images/themes/default/mActionAdd.svg</normaloff>:/images/themes/default/mActionAdd.svg</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QToolButton" name="btnShowBitFlags">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="../resources/bitflagrenderer.qrc">
          <normaloff>:/bitflagrenderer/icons/bitflagmaptool.svg</normaloff>:/bitflagrenderer/icons/bitflagmaptool.svg</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnCopyBitFlagScheme">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../QGIS/images/images.qrc">
          <normaloff>:/images/themes/default/mActionEditCopy.svg</normaloff>:/images/themes/default/mActionEditCopy.svg</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnPasteBitFlagScheme">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../QGIS/images/images.qrc">
          <normaloff>:/images/themes/default/mActionEditPaste.svg</normaloff>:/images/themes/default/mActionEditPaste.svg</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnSaveBitFlagScheme">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../QGIS/images/images.qrc">
          <normaloff>:/images/themes/default/mActionFileSave.svg</normaloff>:/images/themes/default/mActionFileSave.svg</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnLoadBitFlagScheme">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../QGIS/images/images.qrc">
          <normaloff>:/images/themes/default/mActionFileOpen.svg</normaloff>:/images/themes/default/mActionFileOpen.svg</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="5" column="0" colspan="2">
     <widget class="BitFlagRendererTreeView" name="mTreeView">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="acceptDrops">
       <bool>true</bool>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustIgnored</enum>
      </property>
      <property name="dragEnabled">
       <bool>true</bool>
      </property>
      <property name="dragDropOverwriteMode">
       <bool>false</bool>
      </property>
      <property name="dragDropMode">
       <enum>QAbstractItemView::InternalMove</enum>
      </property>
      <property name="alternatingRowColors">
       <bool>false</bool>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::ContiguousSelection</enum>
      </property>
      <property name="autoExpandDelay">
       <number>-1</number>
      </property>
      <property name="rootIsDecorated">
       <bool>true</bool>
      </property>
      <property name="uniformRowHeights">
       <bool>true</bool>
      </property>
      <property name="sortingEnabled">
       <bool>false</bool>
      </property>
      <property name="headerHidden">
       <bool>false</bool>
      </property>
      <attribute name="headerMinimumSectionSize">
       <number>10</number>
      </attribute>
      <attribute name="headerDefaultSectionSize">
       <number>25</number>
      </attribute>
     </widget>
    </item>
    <item row="6" column="0" colspan="2">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="spacing">
       <number>2</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QToolButton" name="btnCombineFlags">
        <property name="text">
         <string>+</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../QGIS/images/images.qrc">
          <normaloff>:/images/themes/default/mIconConnect.svg</normaloff>:/images/themes/default/mIconConnect.svg</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QgsColorButton" name="btnCombinedFlagsColor">
        <property name="color">
         <color>
          <red>170</red>
          <green>170</green>
          <blue>0</blue>
         </color>
        </property>
        <property name="defaultColor">
         <color>
          <red>170</red>
          <green>170</green>
          <blue>0</blue>
         </color>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QCheckBox" name="cbLiveUpdate">
        <property name="text">
         <string>Live update</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnApply">
        <property name="text">
         <string>Apply</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <action name="actionAddParameter">
   <property name="icon">
    <iconset resource="../../../../QGIS/images/images.qrc">
     <normaloff>:/images/themes/default/mActionAdd.svg</normaloff>:/images/themes/default/mActionAdd.svg</iconset>
   </property>
   <property name="text">
    <string>addFlagParameter</string>
   </property>
   <property name="toolTip">
    <string>Add a new parameter</string>
   </property>
  </action>
  <action name="actionRemoveParameters">
   <property name="icon">
    <iconset resource="../../../../QGIS/images/images.qrc">
     <normaloff>:/images/themes/default/mActionRemove.svg</normaloff>:/images/themes/default/mActionRemove.svg</iconset>
   </property>
   <property name="text">
    <string>Remove</string>
   </property>
   <property name="toolTip">
    <string>Removes the selected parameter(s).</string>
   </property>
  </action>
  <action name="actionLoadBitFlagScheme">
   <property name="icon">
    <iconset resource="../../../../QGIS/images/images.qrc">
     <normaloff>:/images/themes/default/mActionFileOpen.svg</normaloff>:/images/themes/default/mActionFileOpen.svg</iconset>
   </property>
   <property name="text">
    <string>Load bit Flag Scheme</string>
   </property>
   <property name="toolTip">
    <string>Loads a bit flag scheme</string>
   </property>
  </action>
  <action name="actionSaveBitFlagScheme">
   <property name="icon">
    <iconset resource="../../../../QGIS/images/images.qrc">
     <normaloff>:/images/themes/default/mActionFileSave.svg</normaloff>:/images/themes/default/mActionFileSave.svg</iconset>
   </property>
   <property name="text">
    <string>Save Bit Flag Scheme</string>
   </property>
   <property name="toolTip">
    <string>Saves the current Bit Flag Scheme</string>
   </property>
  </action>
  <action name="actionCombineFlags">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../../../QGIS/images/images.qrc">
     <normaloff>:/images/themes/default/mIconConnect.svg</normaloff>:/images/themes/default/mIconConnect.svg</iconset>
   </property>
   <property name="text">
    <string>Combine Flags</string>
   </property>
   <property name="toolTip">
    <string>Activate to combine all selected flags states and show them with unified color</string>
   </property>
  </action>
  <action name="actionCopyBitFlagScheme">
   <property name="icon">
    <iconset resource="../../../../QGIS/images/images.qrc">
     <normaloff>:/images/themes/default/mActionEditCopy.svg</normaloff>:/images/themes/default/mActionEditCopy.svg</iconset>
   </property>
   <property name="text">
    <string>Copy Bit Flag Scheme</string>
   </property>
   <property name="toolTip">
    <string>Copy the current Bit Flag Scheme to clipboard.</string>
   </property>
  </action>
  <action name="actionPasteBitFlagScheme">
   <property name="icon">
    <iconset resource="../../../../QGIS/images/images.qrc">
     <normaloff>:/images/themes/default/mActionEditPaste.svg</normaloff>:/images/themes/default/mActionEditPaste.svg</iconset>
   </property>
   <property name="text">
    <string>Paste Bit Flag Scheme</string>
   </property>
   <property name="toolTip">
    <string>Paste a Bit Flag Scheme from clipboard</string>
   </property>
  </action>
  <action name="actionShowBitFlags">
   <property name="icon">
    <iconset resource="../resources/bitflagrenderer.qrc">
     <normaloff>:/bitflagrenderer/icons/bitflagmaptool.svg</normaloff>:/bitflagrenderer/icons/bitflagmaptool.svg</iconset>
   </property>
   <property name="text">
    <string>Show Bit Flags</string>
   </property>
   <property name="toolTip">
    <string>Shows the bit flags at cursor position</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsColorButton</class>
   <extends>QToolButton</extends>
   <header>qgscolorbutton.h</header>
  </customwidget>
  <customwidget>
   <class>QgsDockWidget</class>
   <extends>QDockWidget</extends>
   <header>qgsdockwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsMapLayerComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsmaplayercombobox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsRasterBandComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsrasterbandcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>BitFlagRendererTreeView</class>
   <extends>QTreeView</extends>
   <header>bitflagrenderer.gui.bitflagrenderertreeview</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/bitflagrenderer.qrc"/>
  <include location="../../../../QGIS/images/images.qrc"/>
 </resources>
 <connections/>
</ui>
